### ISSUE TITLE
Clippy warning on missing `Eq` derive

### ISSUE THREAD

**Issue by bgilbert** [Marwes/schemafy#61]
Clippy 1.63.0 added a [warning](https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq) for types that derive `PartialEq` but not `Eq`, and could have derived `Eq`.  As a result, many types generated by schemafy produce Clippy warnings:

```
warning: you are deriving `PartialEq` and can implement `Eq`
  --> src/mod.rs:18:1
   |
18 | / schemafy!(
19 | |     root: Config
20 | |     "schema.json"
21 | | );
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq
   = note: this warning originates in the macro `schemafy` (in Nightly builds, run with -Z macro-backtrace for more info)
```
